tests/test_model_v2.py

```diff
@@ -807,3 +808,74 @@ def test_edmx_association_set_end_by_entity_set():
 
     assert association_set.end_by_entity_set(end_from.entity_set_name) == end_from
     assert association_set.end_by_entity_set(end_to.entity_set_name) == end_to
+
+
+def test_missing_data_service(metadata_builder_factory):
+    """Test correct handling of missing DataService tag in xml"""
+
+    builder = metadata_builder_factory()
+    builder.data_services_is_enabled = False
+    xml = builder.serialize()
+
+    try:
+        Edmx.parse(xml)
+    except PyODataParserError as ex:
+        assert str(ex) == 'Metadata document is missing the element DataServices'
+
+
+def test_missing_schema(metadata_builder_factory):
+    """Test correct handling of missing Schema tag in xml"""
+
+    builder = metadata_builder_factory()
+    builder.schema_is_enabled = False
+    xml = builder.serialize()
+
+    try:
+        Edmx.parse(xml)
+    except PyODataParserError as ex:
+        assert str(ex) == 'Metadata document is missing the element Schema'
+
+
+def test_namespace_whitelist(metadata_builder_factory):
+    """Test correct handling of whitelisted namespaces"""
+
+    builder = metadata_builder_factory()
+    builder.namespaces['edmx'] = 'http://docs.oasis-open.org/odata/ns/edmx'
+    builder.namespaces['edm'] = 'http://docs.oasis-open.org/odata/ns/edm'
+    builder.add_schema('', '')
+    xml = builder.serialize()
+    Edmx.parse(xml)
```

Comment:
There must be some assert or anything that makes sure it went OK or not?

Reply:
The function finishes with no exception raised so, the test passes otherwise it would fail. How can I assert for no exception in pytest to make this explicit?

Reply:
Yep, but even an empty function would succeed, right? So we either need a coverage checker or we need to make sure we actually tested something.

Reply:
Agree. Do you have a tip for how to do it? (I haven't work with pytest before.) I tried to google it but I came up with nothing.

Reply:
Mock the latest possible method call and check it was called:
https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once